chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -207,186 +74,13 @@
-      owner "root"
-      group "root"
-      mode "0644"
-      action :create
-      source "crowbar_notify_shutdown.service"
-    end
-
-    cookbook_file "/etc/systemd/system/crowbar_join.service" do
-      owner "root"
-      group "root"
-      mode "0644"
-      action :create
-      source "crowbar_join.service"
-    end
-
-    # Make sure that any dependency change is taken into account
-    bash "reload systemd after crowbar_join update" do
-      code "systemctl daemon-reload"
-      action :nothing
-      subscribes :run, resources(cookbook_file: "/etc/systemd/system/crowbar_notify_shutdown.service"), :immediately
-      subscribes :run, resources(cookbook_file: "/etc/systemd/system/crowbar_join.service"), :immediately
-    end
-
-    link "/usr/sbin/rccrowbar_join" do
-      action :create
-      to "service"
-    end
-
-    service "crowbar_notify_shutdown" do
-      action [:enable, :start]
-    end
-  end
-
-  service "crowbar_join" do
-    action :enable
-  end
-
-  cookbook_file "/etc/logrotate.d/crowbar_join" do
-    owner "root"
-    group "root"
-    mode "0644"
-    source "crowbar_join.logrotate.suse"
-    action :create
-  end
-
-  # remove old crowbar_join.sh file
-  file "/etc/init.d/crowbar_join.sh" do
-    action :delete
-  end
-
-  if node[:platform_family] == "suse"
-    # make sure the repos are properly setup
-    repos = Provisioner::Repositories.get_repos(node[:platform], node[:platform_version], node[:kernel][:machine])
-    for name, attrs in repos
-      current_url = nil
-      current_priority = nil
-
-      out = `LANG=C zypper --non-interactive repos #{name} 2> /dev/null`
-      out.split("\n").each do |line|
-        attribute, value = line.split(":", 2)
-        next if value.nil?
-        attribute.strip!
-        value.strip!
-        if attribute == "URI"
-          current_url = value
-        elsif attribute == "Priority"
-          current_priority = value
-        end
-      end
-
-      if current_url != attrs[:url]
-        unless current_url.nil? || current_url.empty?
-          Chef::Log.info("Removing #{name} zypper repository pointing to wrong URI...")
-          `zypper --non-interactive removerepo #{name}`
-        end
-        Chef::Log.info("Adding #{name} zypper repository...")
-        `zypper --non-interactive addrepo --refresh #{attrs[:url]} #{name}`
-      end
-      if current_priority != attrs[:priority]
-        `zypper --non-interactive modifyrepo --priority #{attrs[:priority]} #{name}`
-      end
-    end
-    # install additional packages
-    os = "#{node[:platform]}-#{node[:platform_version]}"
-    if node[:provisioner][:packages][os]
-      node[:provisioner][:packages][os].each { |p| package p }
-    end
-  end
-end
-
-aliaz = begin
-  display_alias = node["crowbar"]["display"]["alias"]
-  if display_alias && !display_alias.empty?
-    display_alias
-  else
-    node["hostname"]
-  end
-rescue
-  node["hostname"]
+  include_recipe "provisioner::client"
```

Comment:
Can you move the `if platform == suse` bit inside this new recipe? As the recipe should be run regardless of the platform.